components/menu/menu_item.tsx

```diff
@@ -0,0 +1,208 @@
+     * Wraps the labels with element such as span, div etc. to support styling instead of passing text node directly.
+     */
+    labels: ReactElement;
+
+    /**
+     * The meta data element to display extra information about menu item. Could be chevron, shortcut or badge.
+     * It is formed with subMenuDetail and trailingElement. If only one is passed, it will be tailingElement. If two are
+     * passed, first will be subMenuDetail and second will be trailingElement.
+     *
+     * @example
+     * <Menu.Item trailingElements={<ChevronRightIcon/>}/>
+     *
+     * @example
+     * <Menu.Item trailingElements={
+     *  <>
+     *   <FormattedMessage id="submenu.detail" defaultMessage="submenu detail"/>
+     *   <ChevronRightIcon/>
+     * </>
+     * }/>
+     */
+    trailingElements?: ReactNode;
+
+    /**
+     * For actions of menu item that are destructive in nature and harder to undo.
+     */
+    isDestructive?: boolean;
+
+    /**
+     * ONLY to support submenus. Avoid passing children to this component. Support for children is only added to support submenus.
+     */
+    children?: ReactNode;
+}
+
+/**
+ * To be used as a child of Menu component.
+ * Checkout Compass's Menu Item(compass.mattermost.com)  for terminology, styling and usage guidelines.
+ *
+ * @example
+ * <Menu.Container>
+ *    <Menu.Item/>
+ * </Menu.Container>
+ */
+export function MenuItem(props: Props) {
+    const {
+        leadingElement,
+        labels,
+        trailingElements,
+        isDestructive,
+        children,
+        ...restProps
+    } = props;
+
+    // When both primary and secondary labels are passed, we need to apply minor changes to the styling. Check below in styled component for more details.
+    const hasSecondaryLabel = labels && labels.props && labels.props.children && Children.count(labels.props.children) === 2;
+
+    return (
+        <MenuItemStyled
+            disableRipple={true}
+            disableTouchRipple={true}
+            isDestructive={isDestructive}
+            hasSecondaryLabel={hasSecondaryLabel}
+            {...restProps}
+        >
+            {leadingElement && <div className='leading-element'>{leadingElement}</div>}
+            <div className='label-elements'>{labels}</div>
+            {trailingElements && <div className='trailing-elements'>{trailingElements}</div>}
+            {children}
+        </MenuItemStyled>
+    );
+}
+
+interface MenuItemStyledProps extends MuiMenuItemProps {
+    isDestructive?: boolean;
+    hasSecondaryLabel?: boolean;
+}
+
+/* eslint-disable no-negated-condition */
+const MenuItemStyled = styled(MuiMenuItem, {
+    shouldForwardProp: (prop) => prop !== 'isDestructive' && prop !== 'hasSecondaryLabel',
+})<MenuItemStyledProps>(
+    ({isDestructive = false, hasSecondaryLabel = false}) => ({
+        '&.MuiMenuItem-root': {
+            fontFamily: '"Open Sans", sans-serif',
+            color: !isDestructive ? 'var(--center-channel-color)' : 'var(--error-text)',
```

Comment:
could you turn that condition around ... strange to test for `false` instead of `true`

```suggestion
            color: isDestructive ? 'var(--error-text)' : 'var(--center-channel-color)',
```

Reply:
I would like to prefer, primary first and secondary later.

Reply:
IMHO it's usually better to defer to ESLint, rather than introducing `eslint-disabled`s unnecessarily. A big purpose of linting is to avoid these stylistic back and forths.

Reply:
agreed, changed the variable name and removed the eslint disabled